WordPress/Classes/ViewRelated/Post/PrepublishingAutoSharingView.swift

```diff
@@ -2,59 +2,68 @@ import SwiftUI
 
 struct PrepublishingAutoSharingView: View {
 
-    // TODO: view model.
+    let model: PrepublishingAutoSharingViewModel
 
     var body: some View {
         HStack {
             textStack
             Spacer()
-            iconTrain
+            if model.connections.count > 0 {
+                socialIconsView
+            }
         }
     }
 
-    var textStack: some View {
-        VStack(alignment: .leading) {
-            Text(String(format: Strings.primaryLabelActiveConnectionsFormat, 3))
+    private var textStack: some View {
+        VStack(alignment: .leading, spacing: .zero) {
+            Text(model.labelText)
                 .font(.body)
                 .foregroundColor(Color(.label))
-            Text(String(format: Strings.remainingSharesTextFormat, 27, 30))
-                .font(.subheadline)
-                .foregroundColor(Color(.secondaryLabel))
+            if let sharingLimit = model.sharingLimit {
+                remainingSharesView(sharingLimit: sharingLimit, showsWarning: model.showsWarning)
+            }
         }
     }
 
-    // TODO: This will be implemented separately.
-    var iconTrain: some View {
-        HStack {
-            if let uiImage = UIImage(named: "icon-tumblr") {
-                Image(uiImage: uiImage)
+    private func remainingSharesView(sharingLimit: PublicizeInfo.SharingLimit, showsWarning: Bool) -> some View {
+        HStack(spacing: 4.0) {
+            if showsWarning {
+                Image("icon-warning")
                     .resizable()
-                    .frame(width: 32.0, height: 32.0)
-                    .background(Color(UIColor.listForeground))
-                    .clipShape(Circle())
-                    .overlay(Circle().stroke(Color(UIColor.listForeground), lineWidth: 2.0))
+                    .frame(width: 16.0, height: 16.0)
+                    .padding(4.0)
+            }
+            Text(String(format: Constants.remainingSharesTextFormat, sharingLimit.remaining, sharingLimit.limit))
+                .font(.subheadline)
+                .foregroundColor(Color(showsWarning ? Constants.warningColor : .secondaryLabel))
+        }
+    }
+
+    private var socialIconsView: some View {
+        HStack(spacing: -2.0) {
+            ForEach(model.connections, id: \.self) { connection in
+                iconImage(connection.serviceName.localIconImage, opaque: connection.enabled)
             }
```

Comment:
Is it possible for a user to have so many connections it breaks the view? If so, should we limit the number we display?

![Simulator Screenshot - iPhone 14 Pro - 2023-07-11 at 17 12 52](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/a8e76a45-9f52-4c67-b418-41d1bf295f17)